pydra/engine/core.py

```diff
@@ -441,6 +437,8 @@ def __init__(
                     + [("_graph", ty.Any)],
                     bases=(BaseSpec,),
                 )
+        else:
+            raise Exception("Workflow has to have input_spec")
```

Comment:
I think you can just do this with types:

```Diff
 class Workflow(TaskBase):
     def __init__(
         self,
         name,
-        input_spec: ty.Union[ty.List[ty.Text], BaseSpec, None] = None,
+        *,
+        input_spec: ty.Union[ty.List[ty.Text], BaseSpec],
         output_spec: ty.Optional[BaseSpec] = None,
         audit_flags: AuditFlag = AuditFlag.NONE,
```

Keyword-only arguments don't have to have defaults.